src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionTelemetry.cs

```diff
@@ -60,38 +62,41 @@ protected override void OnEventCommand(EventCommandEventArgs command)
         [NonEvent]
         public long BeforeResolution(object hostNameOrAddress)
         {
-            // System.Diagnostics.Metrics part
-            NameResolutionMetrics.BeforeResolution(hostNameOrAddress, out string? host);
-
-            // System.Diagnostics.Tracing part
-            if (IsEnabled())
+            bool telemetryEnabled = IsEnabled();
+            if (telemetryEnabled)
             {
                 Interlocked.Increment(ref _lookupsRequested);
                 Interlocked.Increment(ref _currentLookups);
 
                 if (IsEnabled(EventLevel.Informational, EventKeywords.None))
                 {
-                    host ??= NameResolutionMetrics.GetHostnameFromStateObject(hostNameOrAddress);
+                    string host = GetHostnameFromStateObject(hostNameOrAddress);
 
                     ResolutionStart(host);
                 }
 
                 return Stopwatch.GetTimestamp();
             }
 
-            return 0;
+            return telemetryEnabled || NameResolutionMetrics.IsEnabled() ? Stopwatch.GetTimestamp() : 0;
```

Comment:
Nit: if `telemetryEnabled` is true, we are already returning above.

Either the `telemetryEnabled` check here or the `return Stopwatch.GetTimestamp()` above can be removed.